kernel/aster-nix/src/fs/ext2/inode.rs

```diff
@@ -1587,6 +1680,7 @@ impl InodeImpl {
     pub fn set_file_perm(&self, perm: FilePerm) {
         let mut inner = self.0.write();
         inner.desc.perm = perm;
+        inner.desc.ctime = now();
```

Comment:
I think we should set the timestamps consistently at the inode level, not in the internal methods such as `InodeImpl::set_file_perm`. By following this rule, we can be sure that the timestamps would never get updated when using any methods of `Inner` or `InodeImpl`. Thus, there will never be associated cost when using methods of `Inner` or `InodeImpl`.

Reply:
Make sense